CoreScriptsRoot/CoreScripts/NotificationScript2.lua

```diff
@@ -371,56 +408,105 @@ local function sendNotifcation(title, text, image, duration, callback, button1Te
 end
-BindableEvent_SendNotification.OnInvoke = function(title, text, image, duration, callback)
-	sendNotifcation(title, text, image, duration, callback)
+BindableEvent_SendNotification.Event:connect(onSendNotification)
+
+local function onSendNotificationInfo(notificationInfo)
+	if UserInputService.VREnabled then
+		--If VR is enabled, notifications will be handled by Modules.VR.NotificationHub
+		return
+	end
+
+	--If we hit this, it means that the newNotificationPath flag is on and still exists.
+	--When the flag is removed, we should also merge its contents into this one.
+	--Right now, calling sendNotification(...) will essentially translate into a sendNotificationInfo(table), which will then be translated back
+	--into a parameter list here. One option in the future is to update the code that produces the notification in non-VR to use notificationInfo table.
+	onSendNotification(notificationInfo.Title, notificationInfo.Text, notificationInfo.Image, notificationInfo.Duration, notificationInfo.Callback, notificationInfo.Button1Text, notificationInfo.Button2Text)
 end
+BindableEvent_SendNotificationInfo.Event:connect(onSendNotificationInfo)
 
 -- New follower notification
 spawn(function()
 	local RobloxReplicatedStorage = game:GetService('RobloxReplicatedStorage')
 	local RemoteEvent_NewFollower = RobloxReplicatedStorage:WaitForChild('NewFollower')
 	--
 	RemoteEvent_NewFollower.OnClientEvent:connect(function(followerRbxPlayer)
-		sendNotifcation("New Follower", followerRbxPlayer.Name.." is now following you!",
-			FRIEND_IMAGE..followerRbxPlayer.userId.."&x=48&y=48", 5, function() end)
+		if newNotificationPath then
+			local message = ("%s is now following you"):format(followerRbxPlayer.Name)
+			local image = FRIEND_IMAGE..followerRbxPlayer.userId.."&x=48&y=48"
+			sendNotificationInfo {
+				GroupName = "Friends",
+				Title = "New Follower",
+				Text = message,
+				DetailText = message,
+				Image = image,
+				Duration = 5
+			}
+		else
+			sendNotification("New Follower", followerRbxPlayer.Name.." is now following you!",
+				FRIEND_IMAGE..followerRbxPlayer.userId.."&x=48&y=48", 5, function() end)
+		end
 	end)
 end)
 
 local function sendFriendNotification(fromPlayer)
-	local notification = {}
-	local notificationFrame = createNotification(fromPlayer.Name, "Sent you a friend request!",
-		FRIEND_IMAGE..tostring(fromPlayer.userId).."&x=48&y=48")
-	notificationFrame.Position = UDim2.new(1, 4, 1, -(NOTIFICATION_Y_OFFSET + 2) * 1.5 - 4)
-	--
-	local acceptButton = createTextButton("AcceptButton", "Accept", UDim2.new(0, 0, 1, 2))
-	acceptButton.Parent = notificationFrame
+	--TODO: remove this flag check when stable
+	if newNotificationPath then
+		local acceptText = "Accept"
+		local declineText = "Decline"
+		sendNotificationInfo {
+			GroupName = "Friends",
+			Title = fromPlayer.Name,
+			Text = "Sent you a friend request!",
+			DetailText = fromPlayer.Name,
+			Image = FRIEND_IMAGE .. tostring(fromPlayer.userId) .. "&x=48&y=48",
```

Comment:
Fromat this as well, perhaps we should make a small function for generating this url from a user or userId